src/urllib3/connection.py

```diff
@@ -101,9 +101,7 @@ class HTTPConnection(_HTTPConnection, object):
 
     default_port = port_by_scheme["http"]
 
-    #: Disable Nagle's algorithm by default.
-    #: ``[(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)]``
-    default_socket_options = [(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)]
+    default_socket_options = []
```

Comment:
This is backwards incompatible. There is certainly code today that clones this list and includes it when passing socket options so that they get the good and reasonable defaults from the library and whatever extra they wish to use.

Reply:
> This is backwards incompatible.

Clearly!

Reply:
=> restored